### PR TITLE
updated project for Elasticsearch 2.1.0 with osmosis 0.44.1

### PR DESCRIPTION
**Pull request by clydedacruz**
Made a few updates for using this with Elasticsearch 2.x

### CODE REVIEW

**Comment by ncolomer** [comment]
Thanks a lot for this submission @clydedacruz 
Here are my reviews...

File: src/main/java/org/openstreetmap/osmosis/plugin/elasticsearch/client/ElasticsearchClientBuilder.java [hunk 1]
```diff
@@ -91,17 +92,22 @@ protected Client buildTransportClient() {
...
+		TransportClient transportClient = TransportClient.builder().settings(settings).build();
 		// Add specified TransportAddresses
 		for (String host : hosts.split(",")) {
 			String[] params = host.split(":");
 			String hostname = params[0];
 			int port = (params.length == 2) ? Integer.valueOf(params[1]) : 9300;
-			transportClient.addTransportAddress(new InetSocketTransportAddress(hostname, port));
+			try {
+				transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(hostname), port));
+			} catch (UnknownHostException e) {
```
Comment (ncolomer) [inline src/main/java/org/openstreetmap/osmosis/plugin/elasticsearch/client/ElasticsearchClientBuilder.java:107]: Why this change? Pass the hostname string parameter did not the trick?

File: src/test/java/org/openstreetmap/osmosis/plugin/elasticsearch/model/entity/ESNodeITest.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public void searchNode_withGeoShape() {
 		ShapeBuilder shape = buildSquareShape(48.675652, 2.384955, 20);
 		SearchResponse searchResponse = client().prepareSearch(INDEX_NAME).setTypes(ESEntityType.NODE.getIndiceName())
 				.setQuery(QueryBuilders.matchAllQuery())
-				.setPostFilter(new GeoShapeFilterBuilder("shape", shape))
+				.setPostFilter(new GeoShapeQueryBuilder("shape", shape))
```
Comment (ncolomer) [inline src/test/java/org/openstreetmap/osmosis/plugin/elasticsearch/model/entity/ESNodeITest.java:72]: Filter to query? huh? :)

**Comment by ncolomer** [comment]
You should also upgrade elasticsearch to 2.1.x which where released recently

File: src/main/java/org/openstreetmap/osmosis/plugin/elasticsearch/client/ElasticsearchClientBuilder.java (hunk 1 above)
Comment (clydedacruz) [inline src/main/java/org/openstreetmap/osmosis/plugin/elasticsearch/client/ElasticsearchClientBuilder.java:107]: in 2.0+ the `InetSocketTransportAddress` constructor looks like :
`org.elasticsearch.common.transport.InetSocketTransportAddress.InetSocketTransportAddress(InetAddress address, int port)`

so it will not automatically cast a String to `InetAddress`

File: src/test/java/org/openstreetmap/osmosis/plugin/elasticsearch/model/entity/ESNodeITest.java (hunk 2 above)
Comment (clydedacruz) [inline src/test/java/org/openstreetmap/osmosis/plugin/elasticsearch/model/entity/ESNodeITest.java:72]: The GeoShape filter has been replaced by GeoShape query 
[query-dsl-geo-shape-filter](https://www.elastic.co/guide/en/elasticsearch/reference/2.0/query-dsl-geo-shape-filter.html)

**Comment by Sparkycz** [comment]
bump.. shall we update it, please? Currently last version of ES is 2.3.3

**Comment by ncolomer** [comment]
Sorry for latency. Honestly, I do not actively maintain this project anymore.
However, I merged your PR into branch [2.1.x](https://github.com/ncolomer/elasticsearch-osmosis-plugin/tree/2.1.x) (now the default one).

**Comment by koyuch** [comment]
hi @ncolomer. could you also build the merged branch? or could you at least describe how to correctly build this plugin? (I've tried it with "mvn build", but it didn't seem to work correctly on launch) thanks

**Comment by ncolomer** [comment]
@koyuch not sure to fully understand your request
All project branches are green according to [CircleCI](https://circleci.com/gh/ncolomer/elasticsearch-osmosis-plugin)
Build procedure is explained in the [How to contribute](https://github.com/ncolomer/elasticsearch-osmosis-plugin/wiki/How-to-contribute#the-devs-corner) wiki page
However, I just realized I forgot to generate the release artifact. It's now done and the [README.md](https://github.com/ncolomer/elasticsearch-osmosis-plugin/blob/2.1.x/README.md) link is not a 404 anymore
